<h3><b>Collection of Bioinformatics Scripts in BASH</b></h3>

The BASH scripts were used in bioinformatical analysis of next-generation sequencing data from DNase I digested genomes. 
  
  <li>Aggregated_Plot: Distinct binding profile of transcription factors was visualized while aggregated DNase I cleavage was plotted to distance from motif centers. 

  <li>Cleavage_Heatmap: Distinct binding profile of transcription factors was visualized while DNase I cleavage was plotted to distance from motif centers on heatmap.  

  <li>Conservation_Plot: Sequence conservation of motif instances was plotted to distance from motif centers. 

  <li>Cut_Counts_and_Density: Frequency of DNase I cleavage was calculated as counts of sequencing tags at each nucleotide, and density file of DNase I cleavage was generated across the genome. 

  <li>DESeq: Sequencing data of multiple microbial cultures were compared with the R package "DESeq" to test for differential sensitivity to DNase I cleavage. 

  <li>FIMO: Genome was scanned with FIMO (Find Individual Motif Occurences) of the MEME Suite to locate potential motif instances of the known transcription factors. 

  <li>Footprints_Location: Footprints were summarized according to their relative distances from the nearest transcription start sites, gene start sites, or sigma factor binding sites. 

  <li>Fragment_VPlot: Distinct binding profile of transcription factors was visualized as V-shape gaps while lengths of paired-end sequencing fragments were plotted to distance from motif centers. 

  <li>Genome_Alignment: Sequencing reads were aligned to unique positions on microbial genome with Burrows-Wheeler Aligner. 

  <li>MEME: Motifs of transcription factors were detected with MEME (Multiple Em for Motif Elicitation) of the MEME Suite, the motif-based sequence analysis tools. 

  <li>SPP: DNase I hypersensitive sites were located with the R package "SPP" as regions condensed with sequencing tags. 
